tests/raftstore/test_transport.rs

```diff
@@ -0,0 +1,107 @@
+use super::cluster::{Cluster, Simulator};
+use super::node::new_node_cluster;
+use super::server::new_server_cluster;
+use std::collections::HashSet;
+
+fn split_leader_with_majority(leader: u64, count: u64) -> (HashSet<u64>, HashSet<u64>) {
+    let mut s1 = HashSet::new();
```

Comment:
We can just test if leader is greater than `count / 2 + 1`. If it does, split `1..count + 1` with `(count + 1) / 2`, otherwise split with `count / 2 + 2`.